### PR TITLE
Cascading npmrc files

### LINKED ISSUE #67: Improve robustness of .npmrc imports
Some or all of this may be YAGNI, but just pointing out what I've noticed.

1. Only user path is checked, out of the 4 possible locations support by npm at this time
2. Assumes only one file exists, no "cascading" at a per-property level
3. Vulnerable to changes in precedence or locations in future npm version
4. Vulnerable to changes in format in future npm versions

My solution to all of the above would be to shell out to `npm config get ..` if it exists, then the only brittleness is that API changing (doubtful). The existing behaviour could be maintained as a fallback (potentially updated to handle 1 and 2 above)

@guybedford

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes #67

Searches through the project, user, global and built-in npmrc locations in order.

Namespace keys will be searched in all files before the non-namespaced version is looked up.

### CODE REVIEW

File: lib/npmrc.js [hunk 1]
```diff
@@ -2,23 +2,42 @@ var asp = require('rsvp').denodeify;
 var fs = require('graceful-fs');
 var path = require('path');
 var utils = require('./utils');
+var which = require('which');
 
-function getFilepath() {
+function getFilepaths() {
   var home = process.env.HOME || process.env.USERPROFILE || process.env.HOMEPATH;
```
Comment (guybedford) [inline lib/npmrc.js]: This should be `isWindows ? process.env.USERPROFILE : process.env.HOME` as in https://github.com/npm/osenv/blob/master/osenv.js#L53.

File: lib/npmrc.js [hunk 2]
```diff
@@ -2,23 +2,42 @@ var asp = require('rsvp').denodeify;
...
-  return path.resolve(home, '.npmrc');
+  var paths = ['.npmrc', path.resolve(home, '.npmrc')];
+
+  var prefix = process.env.PREFIX;
```
Comment (guybedford) [inline lib/npmrc.js]: Perhaps this is overkill?

**Comment by guybedford** [comment]
Just reading up on https://docs.npmjs.com/files/npmrc for the first time, and wondering if any of this is even necessary?

**Comment by crisptrutski** [comment]
I'd say that project specific npmrc files are valuable. The others seem more exotic (in latest npm, the builtin does not affect us)

**Comment by guybedford** [comment]
Ok, so we just need to determine the project root then? How are we doing this currently?

**Comment by crisptrutski** [comment]
Assuming process directory is project root seems sane to me. Or directory containing config.js

File: lib/npmrc.js (hunk 1 above)
Comment (crisptrutski) [inline lib/npmrc.js]: Sure

**Pull request merged**